--!strict
export type Data = {
	Id: string,
	VehicleCompatibility: string,
	Name: string,
	Manufacturer: string,
	Model: string,
	Description: string,
	MaxSpeed: number,
	MaxSpeedCarCount: number,
	FuelCostPerKilowatt: number,
	AirPressure: number,
	BaseEngineWeight: number,
	BaseCarWeight: number,
	MaxCarCargoWeight: number,
	CarCargoVolume: number,
	CarPassengerLimit: number,
	FrictionCoefficient: number,
	DragCoefficient: number,
	Lifespan: number,
	ConstructionCost: number,
	MaintenanceCost: number,
}

export type DataKeys = "Callisto" | "Denver" | "Pacifique" | "Gulf" | "Magnus" | "Gigapod" | "Marauder"
return {
	["Callisto"] = {
		["Id"] = "Rail_Callisto",
		["VehicleCompatibility"] = "Rail",
		["Name"] = "Callisto",
		["Manufacturer"] = "Lafayette Locomotive Works",
		["Model"] = "No. 119",
		["Description"] = "The Callisto is classic American steam locomotive, using immense mechanical forces to slowly pull massive amounts of cargo across the nation.",
		["MaxSpeed"] = 50,
		["MaxSpeedCarCount"] = 16,
		["FuelCostPerKilowatt"] = 0.03439380911,
		["AirPressure"] = 1.0,
		["BaseEngineWeight"] = 50,
		["BaseCarWeight"] = 25.0,
		["MaxCarCargoWeight"] = 100,
		["CarCargoVolume"] = 75,
		["CarPassengerLimit"] = 0,
		["FrictionCoefficient"] = 0.0035,
		["DragCoefficient"] = 0.8,
		["Lifespan"] = 30,
		["ConstructionCost"] = 1000,
		["MaintenanceCost"] = 33.33333333,
	},
	["Denver"] = {
		["Id"] = "Rail_Denver",
		["VehicleCompatibility"] = "Rail",
		["Name"] = "Denver",
		["Manufacturer"] = "Monarch",
		["Model"] = "E7-F",
		["Description"] = "The Denver is an iconic American streamliner locomotive known for providing decades of reliable service.",
		["MaxSpeed"] = 70,
		["MaxSpeedCarCount"] = 30,
		["FuelCostPerKilowatt"] = 0.2699914015,
		["AirPressure"] = 1.0,
		["BaseEngineWeight"] = 100,
		["BaseCarWeight"] = 25.0,
		["MaxCarCargoWeight"] = 100,
		["CarCargoVolume"] = 75,
		["CarPassengerLimit"] = 0,
		["FrictionCoefficient"] = 0.0035,
		["DragCoefficient"] = 0.95,
		["Lifespan"] = 40,
		["ConstructionCost"] = 2000,
		["MaintenanceCost"] = 50.0,
	},
	["Pacifique"] = {
		["Id"] = "HighSpeedRail_Pacifique",
		["VehicleCompatibility"] = "HighSpeedRail",
		["Name"] = "Pacifique",
		["Manufacturer"] = "SNCR",
		["Model"] = "T-1988",
		["Description"] = "The Pacifique is a world famous European HSR allowing for quick transportation between countries at affordable prices.",
		["MaxSpeed"] = 160,
		["MaxSpeedCarCount"] = 12,
		["FuelCostPerKilowatt"] = 0.2699914015,
		["AirPressure"] = 1.0,
		["BaseEngineWeight"] = 75,
		["BaseCarWeight"] = 25.0,
		["MaxCarCargoWeight"] = 100,
		["CarCargoVolume"] = 75,
		["CarPassengerLimit"] = 0,
		["FrictionCoefficient"] = 0.0035,
		["DragCoefficient"] = 0.3,
		["Lifespan"] = 25,
		["ConstructionCost"] = 2500,
		["MaintenanceCost"] = 100.0,
	},
	["Gulf"] = {
		["Id"] = "Monorail_Gulf",
		["VehicleCompatibility"] = "Monorail",
		["Name"] = "Gulf",
		["Manufacturer"] = "Walt Motors",
		["Model"] = "D71",
		["Description"] = "The Gulf is a retro-futuristic vision for urban transit, however it is also very capable in dealing with high incline mountainous regions due to how it grips the track.",
		["MaxSpeed"] = 180,
		["MaxSpeedCarCount"] = 8,
		["FuelCostPerKilowatt"] = 0.2024935512,
		["AirPressure"] = 1.0,
		["BaseEngineWeight"] = 75,
		["BaseCarWeight"] = 25.0,
		["MaxCarCargoWeight"] = 100,
		["CarCargoVolume"] = 75,
		["CarPassengerLimit"] = 0,
		["FrictionCoefficient"] = 0.004375,
		["DragCoefficient"] = 0.5,
		["Lifespan"] = 30,
		["ConstructionCost"] = 5000,
		["MaintenanceCost"] = 166.6666667,
	},
	["Magnus"] = {
		["Id"] = "MaglevRail_Magnus",
		["VehicleCompatibility"] = "MaglevRail",
		["Name"] = "Magnus",
		["Manufacturer"] = "Magurebu",
		["Model"] = "TK46",
		["Description"] = "The Magnus is a record-breaking Japanese maglev that serves as the backbone for fast transit between densely populated cities. The 'Magnus' branding is an English localization to help market the technology overseas.",
		["MaxSpeed"] = 600,
		["MaxSpeedCarCount"] = 12,
		["FuelCostPerKilowatt"] = 0.06749785039,
		["AirPressure"] = 1.0,
		["BaseEngineWeight"] = 50,
		["BaseCarWeight"] = 25.0,
		["MaxCarCargoWeight"] = 100,
		["CarCargoVolume"] = 75,
		["CarPassengerLimit"] = 0,
		["FrictionCoefficient"] = 0.00035,
		["DragCoefficient"] = 0.2,
		["Lifespan"] = 25,
		["ConstructionCost"] = 5000,
		["MaintenanceCost"] = 200.0,
	},
	["Gigapod"] = {
		["Id"] = "HyperloopRail_Gigapod",
		["VehicleCompatibility"] = "HyperloopRail",
		["Name"] = "Gigapod",
		["Manufacturer"] = "Sturgeon",
		["Model"] = "TT20-F",
		["Description"] = "The Gigapod is about speed above everything else, including fiscal responsibility.",
		["MaxSpeed"] = 1200,
		["MaxSpeedCarCount"] = 12,
		["FuelCostPerKilowatt"] = 0.03374892519,
		["AirPressure"] = 0.2,
		["BaseEngineWeight"] = 50,
		["BaseCarWeight"] = 25.0,
		["MaxCarCargoWeight"] = 100,
		["CarCargoVolume"] = 75,
		["CarPassengerLimit"] = 0,
		["FrictionCoefficient"] = 0.00035,
		["DragCoefficient"] = 0.1,
		["Lifespan"] = 15,
		["ConstructionCost"] = 10000,
		["MaintenanceCost"] = 666.6666667,
	},
	["Marauder"] = {
		["Id"] = "Road_Marauder",
		["VehicleCompatibility"] = "Road",
		["Name"] = "Marauder",
		["Manufacturer"] = "Revolvo",
		["Model"] = "FH",
		["Description"] = "The Marauder is an 'Merican Truck made in 'Merica with a tough name to show other trucking companies that you broke the mold and are really cool and probably a rebel and you don't take orders from nobody.",
		["MaxSpeed"] = 90,
		["MaxSpeedCarCount"] = 1,
		["FuelCostPerKilowatt"] = 0.2699914015,
		["AirPressure"] = 1.0,
		["BaseEngineWeight"] = 12,
		["BaseCarWeight"] = 17.5,
		["MaxCarCargoWeight"] = 15,
		["CarCargoVolume"] = 50,
		["CarPassengerLimit"] = 0,
		["FrictionCoefficient"] = 0.0105,
		["DragCoefficient"] = 0.5,
		["Lifespan"] = 15,
		["ConstructionCost"] = 70,
		["MaintenanceCost"] = 4.666666667,
	},
} :: {[DataKeys]: Data}